File: python/n405_ToHex.py
# https://leetcode-cn.com/problems/convert-a-number-to-hexadecimal/

class Solution:
    def toHex(self, num: int) -> str:
        hex_str = "0123456789abcdef"
        num &= 0xffffffff
        mask = 0b1111
        res = ""
        while num > 0:
            res += hex_str[num & mask]
            num >>= 4
        return res[::-1] if res else "0"
        
        
if __name__ == "__main__":
    solu = Solution()
    print(solu.toHex(-26))

File: cpp/n405_ToHex.cpp
// https://leetcode-cn.com/problems/convert-a-number-to-hexadecimal/
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    string toHex(int num) {
        if (num == 0) return "0";
        string hex_str = "0123456789abcdef";
        unsigned int num2 = num;
        int mask = 0b1111;
        string res = "";
        while (num2 > 0){
            char tmp = hex_str[num2 & mask];
            res += tmp;
            num2 >>= 4;
        }
        reverse(res.begin(), res.end());
        return res;
    }
};

int main(){
    return 0;
}